topdown/casts.go

```diff
@@ -23,7 +23,13 @@ func builtinToNumber(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term
 	case ast.Number:
 		return iter(ast.NewTerm(a))
 	case ast.String:
-		_, err := strconv.ParseFloat(string(a), 64)
+		strValue := string(a)
+
+		if strValue == "Inf" || strValue == "Infinity" || strValue == "NaN" {
```

Comment:
From the docs linked in the issue:

> ParseFloat recognizes the string "NaN", and the (possibly signed) strings "Inf" and "Infinity" as their respective special floating point values. It ignores case when matching. 

So I think we'll need to match ignoring case here

Reply:
I've made the change. Let me know if any other change needs to be made